### PR TITLE
Add convenient way to retrieve user's email after authentication

### PR DESCRIPTION
**Pull request by lchapo**
This change adds the authenticated user's email address as an attribute of the auth class. Doing so makes it easy to reference the user's email in a dash app.

### CODE REVIEW

**Comment by joshbode** [comment]
Ahh - that's a great idea - I've been doing something similar with the `id_token`:
```
from flask import session
import jwt
...
        print(jwt.decode(session['google_oauth_token']['id_token'], verify=False)['email'])
...
```
which is slightly more complicated, but has the benefit that it doesn't involve another request (e.g. to `/oauth2/v2/userinfo`).

One thing I'm not sure about (and _might_ not matter in practice) - this is setting `self.email` for the last user on the single `GoogleOAuth` object for the entire app - might this cause some concurrency issues?

**Comment by joshbode** [comment]
I was able to get the wrong email showing between two concurrent sessions with different accounts up by adding a sleep to the callback:
```
@app.callback(
    Output('welcome', 'children'),
    [Input('placeholder', 'value')]
)
def on_load(value):
    from time import sleep
    sleep(1)
    return "Welcome, {}!".format(auth.email)
```

**Comment by joshbode** [comment]
You can probably use something like `session` to store this info

http://flask.pocoo.org/docs/1.0/api/#flask.session

**Comment by lchapo** [comment]
Great call on the concurrency issue. `session` seems like it should work. That being said, I was having a hard time reproducing the concurrency issue with the old method, so let me know if you're still seeing it using `session`.

File: dash_google_auth/google_oauth.py [hunk 1]
```diff
@@ -35,6 +36,7 @@ def is_authorized(self):
         assert resp.ok, resp.text
 
         email = resp.json()["email"]
+        session['email'] = email
```
Comment (joshbode) [inline dash_google_auth/google_oauth.py]: I'd probably use `get` here to save a `KeyError` if the response is corrupted (which would also 500) and also, I'd save a line :)
```
email = session['email'] = resp.json().get('email')
```

**Comment by joshbode** [comment]
How you've got it now it is basically how I changed locally it once I saw the concurrency issue. I believe that'll address it :)

File: dash_google_auth/google_oauth.py (hunk 1 above)
Comment (lchapo) [reply dash_google_auth/google_oauth.py]: yep, good idea

**Pull request merged**